python/ray/dag/compiled_dag_node.py

```diff
@@ -1459,7 +1564,21 @@ def _build_execution_schedule(
         )
         # Step 2: Generate an execution schedule for each actor using topological sort
         actor_to_execution_schedule = _generate_actor_to_execution_schedule(graph)
-        return actor_to_execution_schedule
+
+        # Step 3: Optimize the execution schedule if configured
+        actor_to_optimized_schedule = _optimize_execution_schedule(
```

Comment:
`_optimize_execution_schedule` is very generic, it would be better to rename to indicate that this is only for overlapping purposes. Similarly, maybe use `overlapped_schedule` instead of `optimized_schedule`.